spec.bs

```diff
@@ -13,3 +13,543 @@ Group: patcg
+  </div>
+
+  <h3 id="the-observe-browsing-topics-http-response-header-header">The \`<code>Observe-Browsing-Topics</code>\` HTTP response header</h3>
+
+  The \`<dfn export http-header><code>Observe-Browsing-Topics</code></dfn>\` HTTP response header can be used to record a caller's topics observation.
+
+  <div algorithm>
+    To <dfn>handle topics response</dfn>, given a [=response=] |response| and a [=request=] request:
+
+    1. If |request|'s [=request/header list=] does not [=list/contain=] [:Sec-Browsing-Topics:] (implying the |request|'s [=request/current URL=] is not eligible for topics), then return.
+    1. Let |topLevelDocument| be |request|'s [=request/window=]'s [=environment settings object/global object=]'s [=Window/navigable=]'s [=navigable/top-level traversable=]'s [=navigable/active document=].
+    1. Let |callerOrigin| be |request|'s [=request/current URL=]'s [=url/origin=].
+    1. Let |list| be |response|'s [=response/header list=].
+    1. Let |headerValues| be the result of [=header list/getting=] [:Observe-Browsing-Topics:] from |list|.
+    1. If |headerValues|'s [=list/size=] is not 1, then return.
+    1. Let |headerValue| be |headerValues|[0].
+    1. Let |observe| be the result of <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-parsing-a-boolean">parsing</a> |headerValue| into Structured Fields <a href="https://www.rfc-editor.org/rfc/rfc8941.html#name-booleans">Boolean</a>.
+    1. If |observe| is true:
+        1. Run the [=collect page topics calculation input data=] algorithm with |topLevelDocument| as input.
+        1. Run the [=collect topics caller origin=] algorithm with |topLevelDocument| and |callerOrigin| as input.
+  </div>
+
+
+  <h3 id="modification-to-http-fetch-steps-header">Modification to HTTP fetch steps</h3>
+  The following step will be added to the [=HTTP fetch=] steps, before checking the redirect status (i.e. "If |actualResponse|’s status is a redirect status, ..."):
+
+  1. [=Handle topics response=], given [=response=] |actualResponse| and [=request=] |request| as input.
+
+  <span class=XXX>TODO: make the modification directly to the fetch spec.</span>
+
+</section>
+
+<section>
+  <h2 id="permissions-policy-integration-header">Permissions policy integration</h2>
+
+  <p>This specification defines a [=policy-controlled feature=] identified by the string
+  "<code><dfn id=browsing-topics-policy-controlled-feature>browsing-topics</dfn></code>". Its <a>default allowlist</a> is <code>*</code>.
+
+  <p>For backward compatibility, this specification also defines a [=policy-controlled feature=] identified by the string
+  "<code><dfn id=interest-cohort-policy-controlled-feature>interest-cohort</dfn></code>". Its <a>default allowlist</a> is <code>*</code>.
+
+</section>
+
+<section>
+  <h2 id="meeting-the-privacy-goals-header">Meeting the privacy goals</h2>
+
```

Comment:
Here is my suggested text for this section:

The Topics API attempts to provide just enough relevant interest information for advertisers to be able to personalize their ads for the user while maintaining user privacy. Some privacy safeguards include: usage in secure contexts only, topic limitation to a human curated taxonomy, different topics given to different sites in the same epoch to prevent cross-site reidentification, noised topics, a limited number of topics provided per epoch, user opt outs, site opt outs, and a suggestion that user agents provide UX to give users choice in which Topics are returned.